vllm/worker/model_runner.py

```diff
@@ -171,6 +203,9 @@ def _prepare_prompt(
         prefix_block_tables: List[List[int]] = []
         multi_modal_input_list: List[torch.Tensor] = []
 
+        if len(seq_group_metadata_list) == 0:
+            return [], [], None, [], [], [], [], set(), None, []
```

Comment:
Let's wrap this with `PreparePromptMetadata`? Also why do we need a special check here?

Reply:
it is much simpler this way otherwise we should make code below to handle seq_group_metadata_list == 0 (there's additional logic outside iterating seq group metadata)